### PR TITLE
(PUP-7042) Mark strings in network

### PR DESCRIPTION
**Pull request by Magisus**
This commit marks user-facing error and info strings in
`lib/puppet/network/*` for translation.

### CODE REVIEW

**Comment by Magisus** [comment]
Closing and reopening to fix CI.

File: lib/puppet/network/auth_config_parser.rb [hunk 1]
```diff
@@ -49,23 +49,23 @@ def parse_rights
...
-      modify_right(right, :deny, value, "denying %s access", count)
-    when "allow_ip"
-      modify_right(right, :allow_ip, value, "allowing IP %s access", count)
-    when "deny_ip"
-      modify_right(right, :deny_ip, value, "denying IP %s access", count)
-    when "method"
-      modify_right(right, :restrict_method, value, "allowing 'method' %s", count)
-    when "environment"
-      modify_right(right, :restrict_environment, value, "adding environment %s", count)
+    when _("allow")
```
Comment (McdonaldSeanp) [inline lib/puppet/network/auth_config_parser.rb]: Should these be translated? The fact that they are used in a case statement makes me think they might not be external facing?

**Comment by McdonaldSeanp** [comment]
👍

File: lib/puppet/network/auth_config_parser.rb (hunk 1 above)
Comment (Magisus) [inline lib/puppet/network/auth_config_parser.rb]: whoops, yeah I need to remove those

File: lib/puppet/network/auth_config_parser.rb [hunk 2]
```diff
@@ -28,11 +28,11 @@ def parse_rights
         right = rights.newright(name, count, @file)
       when /^\s*(allow(?:_ip)?|deny(?:_ip)?|method|environment|auth(?:enticated)?)\s+(.+?)(\s*#.*)?$/
         if right.nil?
-          raise Puppet::ConfigurationError, "Missing or invalid 'path' before right directive at line #{count} of #{@file}"
+          raise Puppet::ConfigurationError, _("Missing or invalid 'path' before right directive at line %{count} of %{file}") % { count: count, file: @file }
```
Comment (MosesMendoza) [inline lib/puppet/network/auth_config_parser.rb]: Can you add a note that `path` is a reserved keyword here (referring to an entry in the authorization config file) and should not be translated?

File: lib/puppet/network/format_support.rb [hunk 3]
```diff
@@ -11,19 +11,19 @@ module ClassMethods
     def convert_from(format, data)
       get_format(format).intern(self, data)
     rescue => err
-      raise Puppet::Network::FormatHandler::FormatError, "Could not intern from #{format}: #{err}", err.backtrace
+      raise Puppet::Network::FormatHandler::FormatError, _("Could not intern from %{format}: %{err}") % { format: format, err: err }, err.backtrace
```
Comment (McdonaldSeanp) [inline lib/puppet/network/format_support.rb]: All the translated strings in this file (format_support.rb) have the same keyword problem, which is they are descriptive names for functions, but technically they are keywords. Should they be left out of translations?

File: lib/puppet/network/formats.rb [hunk 4]
```diff
@@ -30,7 +30,7 @@ def intern(klass, text)
   def intern_multiple(klass, text)
     data = YAML.load(text)
     unless data.respond_to?(:collect)
-      raise Puppet::Network::FormatHandler::FormatError, "Serialized YAML did not contain a collection of instances when calling intern_multiple"
+      raise Puppet::Network::FormatHandler::FormatError, _("Serialized YAML did not contain a collection of instances when calling intern_multiple")
```
Comment (McdonaldSeanp) [inline lib/puppet/network/formats.rb]: YAML is a keyword, should it be left out of translation?

File: lib/puppet/network/formats.rb [hunk 5]
```diff
@@ -42,7 +42,7 @@ def data_to_instance(klass, data)
     return data if data.is_a?(klass)
 
     unless data.is_a? Hash
-      raise Puppet::Network::FormatHandler::FormatError, "Serialized YAML did not contain a valid instance of #{klass}"
+      raise Puppet::Network::FormatHandler::FormatError, _("Serialized YAML did not contain a valid instance of %{klass}") % { klass: klass }
```
Comment (McdonaldSeanp) [inline lib/puppet/network/formats.rb]: YAML is a keyword, should it be left out of translation?

File: lib/puppet/network/http/request.rb [hunk 6]
```diff
@@ -24,14 +24,14 @@ def format
       header.gsub!(/\s*;.*$/,'') # strip any charset
       format = Puppet::Network::FormatHandler.mime(header)
       if format.nil?
-        raise "Client sent a mime-type (#{header}) that doesn't correspond to a format we support"
+        raise _("Client sent a mime-type (%{header}) that doesn't correspond to a format we support") % { header: header }
```
Comment (McdonaldSeanp) [inline lib/puppet/network/http/request.rb]: `mime-type` looks like a keyword, should it be left in english?

File: lib/puppet/network/http/request.rb [hunk 7]
```diff
@@ -50,7 +50,7 @@ def response_formatter_for(supported_formats, accepted_formats = headers['accept
 
   def assert_supported_format(format)
     if format.name == :yaml || format.name == :b64_zlib_yaml
-      raise Puppet::Error, "YAML in network requests is not supported. See http://links.puppetlabs.com/deprecate_yaml_on_network"
+      raise Puppet::Error, _("YAML in network requests is not supported. See http://links.puppetlabs.com/deprecate_yaml_on_network")
```
Comment (McdonaldSeanp) [inline lib/puppet/network/http/request.rb]: YAML is a keyword, should we leave a comment to leave it out of translation?

File: lib/puppet/network/http/webrick.rb [hunk 8]
```diff
@@ -21,13 +21,13 @@ def listen(address, port)
 
     @server.mount('/', Puppet::Network::HTTP::WEBrickREST)
 
-    raise "WEBrick server is already listening" if @listening
+    raise _("WEBrick server is already listening") if @listening
```
Comment (McdonaldSeanp) [inline lib/puppet/network/http/webrick.rb]: WEBrick is a keyword, should it be left out of translation?

File: lib/puppet/network/http/webrick.rb [hunk 9]
```diff
@@ -37,7 +37,7 @@ def listen(address, port)
   end
 
   def unlisten
-    raise "WEBrick server is not listening" unless @listening
+    raise _("WEBrick server is not listening") unless @listening
```
Comment (McdonaldSeanp) [inline lib/puppet/network/http/webrick.rb]: WEBrick is a keyword, should it be left out of translation?

**Comment by MosesMendoza** [comment]
hey @McdonaldSeanp  - I replied similarly on #5587, but I think the terms you've mentioned in this pr are ok to leave as-is. The reason being I think they'll be covered by a 'one-time' approach via a translation common terms helper/glossary. that way we don't have to duplicate the same translator comment across every file/instance

**Comment by McdonaldSeanp** [comment]
@MosesMendoza that works for me

File: lib/puppet/network/formats.rb (hunk 4 above)
Comment (Magisus) [inline lib/puppet/network/formats.rb]: I'm going to assume that YAML is the same in all languages and the the translators will know to leave it alone.

Comment (McdonaldSeanp) [inline lib/puppet/network/formats.rb]: Works for me

**Comment by Magisus** [comment]
Again, I updated a couple of the ones that seemed more ambiguous and left the rest.

**Pull request merged**